Lib/dataclasses.py

```diff
@@ -365,7 +365,7 @@ def _field_assign(frozen, name, value, self_name):
     # self_name is what "self" is called in this function: don't
     # hard-code "self", since that might be a field name.
     if frozen:
-        return f'object.__setattr__({self_name},{name!r},{value})'
+        return f'__builtins__["object"].__setattr__({self_name},{name!r},{value})'
```

Comment:
Per your comment: I did not know this was how you had to access `__builtins__` inside `exec`.

Reply:
Well, it is `dict` only if you provide the `globals` argument:

```python
>>> exec('print(type(__builtins__))')
<class 'module'>
>>> exec('print(type(__builtins__))', None, None)
<class 'module'>
>>> exec('print(type(__builtins__))', {}, None)
<class 'dict'>
```

That means that the right thing to do is to include `__builtins__` in the `globals` dict, I suppose. I'll send patch in a minute.